Adafruit Feather nrf52840 variants and UF2

jacobw opened the pull request:
Convert boards to variants and add UF2 support to Express.

jacobw committed:
boards: adafruit feather nrf52840 remove bossac

Remove bossac support for UF2 as it isn't neccessary.

Based on xiao_ble.

Signed-off-by: Jacob Winther <jacob@9.nz>

jacobw committed:
boards: adaftuit feather nrf52840 Kconfig updates

Remove unnecessary Kconfig options.
Simplify configuration by removing custom Kconfig option.
Make Kconfig generic for all boards by moving usb selection to
board_defconfig.

These changes should enable the boards to become variants instead.

Signed-off-by: Jacob Winther <jacob@9.nz>

jacobw committed:
boards: adafruit feather nrf52840 convert to variants

Update both versions of adafruit feather nrf52840 to be variants rather
than separate boards. The aim is to provide a more consistent approach
and to remove some unnecessary Kconfig redundancy.

Signed-off-by: Jacob Winther <jacob@9.nz>

decsny commented:
wasn't there another PR about this, or is this a different set of boards, I'm not finding it

jacobw commented:
> wasn't there another PR about this, or is this a different set of boards, I'm not finding it

Yes sorry forgot to link.

Replaces #76629

nordicjm commented on boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi:
```diff
@@ -0,0 +1,48 @@
+		scratch_partition: partition@da000 {
+			label = "image-scratch";
+			reg = <0x000da000 0x0001e000>;
+		};
```
no scratch partition, make slot0 2 sectors bigger than slot1 and use swap using move

jacobw replied:
Ah yeah, I'd just figured that out for another change and forgot to update this one. Done now.

nordicjm commented on boards/adafruit/feather_nrf52840/adafruit_feather_nrf52840_flash.dtsi:
```diff
@@ -0,0 +1,47 @@
+		};
+		slot0_partition: partition@c000 {
+			label = "image-0";
+			reg = <0x0000C000 0x00076000>;
+		};
+		slot1_partition: partition@82000 {
+			label = "image-1";
+			reg = <0x00082000 0x00076000>;
```
```suggestion
			reg = <0x0000C000 0x00077000>;
		};
		slot1_partition: partition@82000 {
			label = "image-1";
			reg = <0x00082000 0x00075000>;
```

jacobw replied:
I'd copied that from the 52840dk. Does you change mean that the two partitions will be different sizes?

nordicjm replied:
If you use swap using move, you need a sector for the data to be moved up by, and you need space for the swap status fields, which is about a sector, so by making the changes here you get the full 0x65000 for an image, without these changes you get 0x64000

jacobw replied:
Got it, thanks for explaining.

jacobw replied:
I adjusted the start for slot1 to match.

Should the 52840dk be updated with the same changes? Partly asking as I have another PR (#77835) to move this to common.

nordicjm replied:
Think you've updated the wrong board

jacobw committed:
boards: adafruit feather nrf52840 update UF2

Add support for flashing the Adafruit Feather nrf52840 Express
using UF2 over USB.

Also standardise on using the UF2 variant, so add a non UF2 variant
of the Sense board, so it can be flashed using the SWD headers.

Created flash dtsi files for both SWD and UF2. These are actually identical
for many other boards and should probably be moved to common in future.

Signed-off-by: Jacob Winther <jacob@9.nz>

jacobw commented:
@anangl ping due to no activity in a week as per PR escalation process.

I've addressed all of the feedback from @nordicjm, though didn't get approval.

nordicjm commented:
Ping @anangl
